### PR TITLE
Clone Microsoft.OperationInsights to a new version and split APIs to separate files - JEDI URGENT

### PR DESCRIPTION
**Pull request by arolshan**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by MyronFanQiu** [comment]
@VeryEarly During our last discussion, do we need this PR or not anymore? Please confirm.

**Comment by VeryEarly** [comment]
@arolshan 

I don't think we need this since the cluster related feature doesn't have overlap with existing specs

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Clusters.json [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Cluster"
+            },
+            "description": "The parameters required to create or update a Log Analytics cluster."
+          }
+        ],
+        "responses": {
+          "202": {
```
Comment (pilor) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Clusters.json]: >"202" [](start = 10, length = 5)

PUT async should be using 201/200 + provisioningState.  This is a change from the previous api-version  [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creatingupdating-using-put)

**Review by pilor — CHANGES_REQUESTED** [review]
1 comment on a change that was made between api-versions

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Clusters.json (hunk 1 above)
Comment (arolshan) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Clusters.json]: this was changed 2 days ago. Please check Clusters.json in master.

Comment (arolshan) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Clusters.json]: This was changed 2 days ago. I've merged the master so u can see

**Review by pilor — APPROVED** [review]

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Clusters.json (hunk 1 above)
Comment (MyronFanQiu) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Clusters.json]: should it be 201?

File: specification/operationalinsights/resource-manager/readme.md [hunk 2]
```diff
@@ -62,6 +62,22 @@ input-file:
 - Microsoft.OperationalInsights/preview/2019-08-01-preview/OperationalInsights.json
 ```
 
+``` yaml $(tag) == 'package-2020-03-preview'
```
Comment (MyronFanQiu) [inline specification/operationalinsights/resource-manager/readme.md]: Could you change the default tag to this one too?

Comment (MyronFanQiu) [reply specification/operationalinsights/resource-manager/readme.md]: I changed it. Please check it.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Clusters.json [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+          "format": "int64"
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the SKU.",
+          "enum": [
+            "CapacityReservation"
+          ],
+          "x-ms-enum": {
+            "name": "SkuNameEnum",
```
Comment (VeryEarly) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Clusters.json]: this defined differently from the other one which blocked dotnet SDK generation

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Clusters.json [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+      "properties": {
+        "capacity": {
+          "description": "The capacity value",
+          "type": "integer",
+          "format": "int64"
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the SKU.",
+          "enum": [
```
Comment (VeryEarly) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Clusters.json]: this defined differently from the other one which blocked dotnet SDK generation

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Workspaces.json [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+      }
+    }
+  },
+  "definitions": {
+    "Sku": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the SKU.",
+          "enum": [
```
Comment (VeryEarly) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Workspaces.json]: this defined differently from the other one which blocked dotnet SDK generation

File: specification/operationalinsights/resource-manager/readme.md (hunk 2 above)
Comment (arolshan) [reply specification/operationalinsights/resource-manager/readme.md]: good

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Clusters.json (hunk 4 above)
Comment (arolshan) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Clusters.json]: sure

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Workspaces.json (hunk 5 above)
Comment (arolshan) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Workspaces.json]: sure

**Pull request merged**